atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/asts/InitialBucketAssignerStateMachineBootstrapTaskTest.java

```diff
@@ -85,59 +86,96 @@ public void allInitialOrResetLastSweptTimestampAutoClosesInitialBucketFromZeroTo
+                .setInitialStateForBucketAssigner(BucketAssignerState.immediatelyClosing(
+                        0, clampToCoarsePartition(freshTimestampSupplier.get())));
+        inOrder.verify(progress).updateNumberOfShards(128);
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {SweepQueueUtils.RESET_TIMESTAMP, SweepQueueUtils.INITIAL_TIMESTAMP})
+    public void allInitialOrResetLastSweptTimestampStartsBucketFromZeroAndBumpsShardsTo128IfClampedFreshTimestampIsZero(
+            long lastSweptTimestamp) {
+        when(progress.getNumberOfShards()).thenReturn(123);
+        when(progress.getLastSweptTimestamps(allShardsAndStrategies(123)))
+                .thenReturn(allShardsAndStrategiesToSameTimestamp(123, lastSweptTimestamp));
+        freshTimestampSupplier.set(1212);
+
+        task.run();
+
+        InOrder inOrder = inOrder(stateMachineTable, progress);
+        inOrder.verify(stateMachineTable).setInitialStateForBucketAssigner(BucketAssignerState.start(0));
         inOrder.verify(progress).updateNumberOfShards(128);
     }
 
     @Test
     public void
-            allSameLastSweptTimestampNotInitialOrResetCreatesOpenInitialBucketFromThatTimestampAndBumpsShardsTo128() {
+            allSameLastSweptTimestampNotInitialOrResetCreatesOpenInitialBucketFromClampedTimestampAndBumpsShardsTo128() {
         when(progress.getNumberOfShards()).thenReturn(62);
         when(progress.getLastSweptTimestamps(allShardsAndStrategies(62)))
-                .thenReturn(allShardsAndStrategiesToSameTimestamp(62, 12));
+                .thenReturn(allShardsAndStrategiesToSameTimestamp(62, 1248465154));
 
         task.run();
         InOrder inOrder = inOrder(stateMachineTable, progress);
 
-        inOrder.verify(stateMachineTable).setInitialStateForBucketAssigner(BucketAssignerState.start(12));
+        inOrder.verify(stateMachineTable)
+                .setInitialStateForBucketAssigner(BucketAssignerState.start(clampToCoarsePartition(1248465154)));
         inOrder.verify(progress).updateNumberOfShards(128);
     }
 
     @Test
-    public void differentLastSweptTimestampsCreatesOpenInitialBucketFromEarliestTimestampAndBumpsShardsTo128() {
+    public void differentLastSweptTimestampsCreatesOpenInitialBucketFromClampedEarliestTimestampAndBumpsShardsTo128() {
         when(progress.getNumberOfShards()).thenReturn(1);
         when(progress.getLastSweptTimestamps(allShardsAndStrategies(1)))
                 .thenReturn(Map.of(
-                        ShardAndStrategy.of(0, SweeperStrategy.CONSERVATIVE), 123L,
-                        ShardAndStrategy.of(0, SweeperStrategy.THOROUGH), 456L));
-        freshTimestampSupplier.set(123456789L);
+                        ShardAndStrategy.of(0, SweeperStrategy.CONSERVATIVE), 4554654655L,
+                        ShardAndStrategy.of(0, SweeperStrategy.THOROUGH), 5464654665L));
+        freshTimestampSupplier.set(6547464687L);
```

Comment:
not actionable: The more readable way to write this is probably to have multiples of the coarse partition values, though I think this is OK 😅